pkg/network/multus/multus_test.go

```diff
@@ -0,0 +1,100 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2022 Red Hat, Inc.
+ *
+ */
+
+package multus_test
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	v1 "kubevirt.io/api/core/v1"
+
+	"kubevirt.io/kubevirt/pkg/network/multus"
+)
+
+var _ = Describe("Multus", func() {
+	Context("FilterNonDefaultNetworks", func() {
+		type filterNonDefaultNetworksParams struct {
+			inputNetworks          []v1.Network
+			expectFilteredNetworks []v1.Network
+		}
+		DescribeTable("should return the expected networks", func(params filterNonDefaultNetworksParams) {
+			filteredNetworks := multus.FilterNonDefaultNetworks(params.inputNetworks)
+
+			Expect(filteredNetworks).To(Equal(params.expectFilteredNetworks))
+		},
+			Entry("when there are no networks", filterNonDefaultNetworksParams{
+				inputNetworks:          []v1.Network{},
+				expectFilteredNetworks: nil,
+			}),
+			Entry("when there are no non-default networks", filterNonDefaultNetworksParams{
+				inputNetworks: []v1.Network{
+					createPodNetwork("default"),
+				},
+				expectFilteredNetworks: nil,
+			}),
+			Entry("when there is a default Multus network", filterNonDefaultNetworksParams{
+				inputNetworks: []v1.Network{
+					createMultusDefaultNetwork("network1", "default/nad1"),
+				},
+				expectFilteredNetworks: nil,
+			}),
+			Entry("when there are non-default Multus networks", filterNonDefaultNetworksParams{
+				inputNetworks: []v1.Network{
+					createMultusDefaultNetwork("network1", "default/nad1"),
+					createMultusSecondaryNetwork("network2", "default/nad2"),
+					createMultusSecondaryNetwork("network3", "default/nad3"),
+				},
+				expectFilteredNetworks: []v1.Network{
+					createMultusSecondaryNetwork("network2", "default/nad2"),
+					createMultusSecondaryNetwork("network3", "default/nad3"),
+				},
+			}),
+		)
+	})
+})
+
+func createMultusSecondaryNetwork(name, networkName string) v1.Network {
+	return createMultusNetwork(name, networkName, false)
```

Comment:
Why the linter did not complain, I do not know.
But you better name the boolean using a const, to make this readable.

I also do not understand why you need a wrapper if you already have a readable argument.
In fact, having a boolean as an argument is a smell, I would have avoided it and then such a wrapper will make sense.

Reply:
I think hiding the boolean in a wrapper makes it more readable

>In fact, having a boolean as an argument is a smell, I would have avoided it and then such a wrapper will make sense

Can you elaborate what do you mean? what would you suggest?

Reply:
I meant this:
```
func newMultusDefaultNetwork(name, nadName) v1.Network {
    net := newMultusNetwork(name, nadName)
    net.NetworkResource.Multus.Default = true
    return net
}
```

The other one you can leave as `newMultusNetwork` or wrap with a different name (I do not see why it is needed, but it is per taste) and add nothing.

Reply:
ACK, got it, Done